### PR TITLE
Feature/uar 1101 retrieve trust data in journey with trustees & links

### PR DESCRIPTION
**Pull request by AndreKramer7654**
### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/update/overseas.entity.change.controller.ts [hunk 1]
```diff
@@ -47,7 +48,10 @@ export const post = async (req: Request, resp: Response, next: NextFunction) =>
       resetDataForChange(appData);
       redirectUrl = config.WHO_IS_MAKING_UPDATE_URL;
     }
+
+    setExtraData(session, appData);
```
Comment (bbell-companieshouse) [inline src/controllers/update/overseas.entity.change.controller.ts:52]: Will this not affect register journey if its moved here?

File: src/utils/update/trust.model.fetch.ts [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+        return RoleWithinTrustType.GRANTOR;
+      case "5003":
+        return RoleWithinTrustType.SETTLOR;
+      case "5002":
+      default:
+        return RoleWithinTrustType.BENEFICIARY;
+  }
+};
+
+const mapResidentialAddress = (trustee: TrustCorporate, trusteeData: CorporateTrusteeData) => {
```
Comment (bbell-companieshouse) [inline src/utils/update/trust.model.fetch.ts:296]: Change this to mapRegisteredOfficeAddress

File: src/controllers/update/overseas.entity.change.controller.ts (hunk 1 above)
Comment (AndreKramer7654) [reply src/controllers/update/overseas.entity.change.controller.ts:52]: This controller is only for update journey. We need to set everything in the Web session so moved that down.

File: src/utils/update/trust.model.fetch.ts (hunk 2 above)
Comment (AndreKramer7654) [reply src/utils/update/trust.model.fetch.ts:297]: good spot. I'm blaming copilot from now on.

File: test/utils/update/trust.model.fetch.spec.ts [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+    await retrieveTrustData(req, appData);
+
+    expect(mockGetTrustData).toBeCalledTimes(1);
+    expect(mockLoggerInfo).toBeCalledTimes(4);
+    expect(appData.update?.review_trusts).toHaveLength(2);
+    expect(appData.beneficial_owners_individual).toEqual(undefined);
+    expect(appData.beneficial_owners_corporate).toEqual(undefined);
+  });
+
+  test("should fetch and not map trust links in with empty benefitial owners", async () => {
```
Comment (bmcgreevyCH) [inline test/utils/update/trust.model.fetch.spec.ts:488]:
```suggestion
  test("should fetch and not map trust links in with empty beneficial owners", async () => {
```

**Review by bmcgreevyCH — APPROVED** [review]

**Review by bbell-companieshouse — APPROVED** [review]

**Pull request merged**